app/models/pawn.rb

```diff
@@ -1,4 +1,24 @@
 class Pawn < Piece
+
+	#en passant status of self
+	def update_en_passant_status(end_x, end_y)
+		if (self.pos_y - end_y).abs == 2
+			self.update_attributes(en_passant: true)
+		else
+			self.update_attributes(en_passant: false)
+		end
+	end
+
+	#determine if pawn can capture opposing pawn en passant
+	def can_capture_en_passant(opponent_pawn)
+		if opponent_pawn.updated_at == game.updated_at && opponent_pawn.en_passant == true 
+			return true
```

Comment:
Yeah good eye on that, you can't rely on game.updated_at ==
pawn.updated_at.  First of all they might not actually be created at the
same time in which case the update_at would not be the same.  Secondly what
jeff said is correct.

On Thu, Oct 1, 2015 at 8:13 PM, jeffp2007 <notifications@github.com> wrote:

> In app/models/pawn.rb
> <https://github.com/Firehose-Team-Awesome/chess-game/pull/25#discussion_r40981822>
> :
>
> > @@ -1,4 +1,24 @@
> >  class Pawn < Piece
> > +
> > +	#en passant status of self
> > +	def update_en_passant_status(end_x, end_y)
> > +		if (self.pos_y - end_y).abs == 2
> > +			self.update_attributes(en_passant: true)
> > +		else
> > +			self.update_attributes(en_passant: false)
> > +		end
> > +	end
> > +
> > +	#determine if pawn can capture opposing pawn en passant
> > +	def can_capture_en_passant(opponent_pawn)
> > +		if opponent_pawn.updated_at == game.updated_at && opponent_pawn.en_passant == true
> > +			return true
>
> Won't the game be continually updating? If a pawn moved 2 spaces at the
> beginning of the game and we are 5 moves down the road, i don't the
> pawn.updated_at will ever equal game.updated_at.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Firehose-Team-Awesome/chess-game/pull/25/files#r40981822>
> .
>